stdlib/src/collections/counter.mojo

```diff
@@ -76,3 +73,363 @@ struct Counter[V: KeyElement](
             count: The count to store in the Counter.
         """
         self._data[value] = count
+
+    fn __iter__(ref [_]self: Self) -> _DictKeyIter[V, Int, __lifetime_of(self)]:
+        """Iterate over the keyword dict's keys as immutable references.
+
+        Returns:
+            An iterator of immutable references to the Counter values.
+        """
+        return self._data.__iter__()
+
+    fn __len__(self) -> Int:
+        """The number of elements currently stored in the Counter."""
+        return len(self._data)
+
+    fn __bool__(self) -> Bool:
+        """Check if the Counter is empty or not.
+
+        Returns:
+            `False` if the Counter is empty, `True` if there is at least one element.
+        """
+        return len(self).__bool__()
```

Comment:
I think it should be the other way around, as `__len__` is the one we defined here.
```suggestion
        return bool(self.__len__())
```
Actually, this should also be good. I don't know which style should we prefer. @laszlokindrat Do we always use explicit method call in implementations?

```suggestion
        return bool(len(self))
```

Reply:
Done: https://github.com/msaelices/mojo/commit/df2f4b1a905afb1f6ced8fdf6fccc8ec91dc2077